app/gui/src/router.tsx

```diff
@@ -0,0 +1,180 @@
+import {
+  CONFIRM_REGISTRATION_PATH,
+  DASHBOARD_PATH,
+  FORGOT_PASSWORD_PATH,
+  RESET_PASSWORD_PATH,
+  RESTORE_USER_PATH,
+  SETUP_PATH,
+  SUBSCRIBE_PATH,
+  SUBSCRIBE_SUCCESS_PATH,
+} from '#/appUtils'
+import { ErrorBoundary } from '#/components/ErrorBoundary'
+import { Suspense } from '#/components/Suspense'
+import { OpenAppWatcher } from '#/layouts/OpenAppWatcher'
+import { AgreementsModal } from '#/modals/AgreementsModal'
+import { InvitedToOrganizationModal } from '#/modals/InvitedToOrganizationModal'
+import { SetupOrganizationAfterSubscribe } from '#/modals/SetupOrganizationAfterSubscribe'
+import ConfirmRegistration from '#/pages/authentication/ConfirmRegistration'
+import ForgotPassword from '#/pages/authentication/ForgotPassword'
+import LoadingScreen from '#/pages/authentication/LoadingScreen'
+import Login from '#/pages/authentication/Login'
+import Registration from '#/pages/authentication/Registration'
+import ResetPassword from '#/pages/authentication/ResetPassword'
+import RestoreAccount from '#/pages/authentication/RestoreAccount'
+import { Setup } from '#/pages/authentication/Setup'
+import Dashboard from '#/pages/dashboard/Dashboard'
+import { Subscribe } from '#/pages/subscribe/Subscribe'
+import { SubscribeSuccess } from '#/pages/subscribe/SubscribeSuccess'
+import {
+  AnyLoggedInUserLayout,
+  CloudBrowserDisabledLayout as CloudBrowserDisabledLayoutImpl,
+  GuestLayout,
+  NotDeletedUserLayout,
+  ProtectedLayout,
+  SoftDeletedUserLayout,
+} from '#/providers/AuthProvider'
+import { PropsWithChildren, ReactNode } from 'react'
+import { applyPureReactInVue } from 'veaury'
+import { createRouter, createWebHistory, RouteRecordRaw } from 'vue-router'
+import ReactLayoutWrapper from './components/ReactLayoutWrapper.vue'
+
+/**
+ * Wrap react component in ErrorBoundary and Suspense.
+ *
+ * The Router views doesn't really like things thrown at them from react.
+ */
+function wrapReactForRouter(Component: (props: PropsWithChildren) => ReactNode) {
+  return ({ children }: PropsWithChildren) => (
+    <ErrorBoundary>
+      <Suspense fallback={<LoadingScreen />}>
+        <Component>{children}</Component>
+      </Suspense>
+    </ErrorBoundary>
+  )
+}
+
+function reactForRouter(component: () => ReactNode) {
+  return applyPureReactInVue(wrapReactForRouter(component))
+}
+
+/**
+ * Nests multiple ReactLayouts and create a Route.
+ *
+ * This function avoids making too many vue-react boundaries.
+ */
+function applyLayouts(
+  components: ((props: PropsWithChildren) => ReactNode)[],
+  children: RouteRecordRaw[],
+) {
+  const reducedComponent = components.reduceRight(
+    (Composed, Next) =>
+      ({ children }: PropsWithChildren) => (
+        <Next>
+          <Composed>{children}</Composed>
```

Comment:
This creates an extra functional component for each reduced layer. I think It could be simplified to just one layer by calling the reduced component function instead of using it in JSX.
```jsx
(composed, Next) => (props) => <Next>{composed(props)}</Next>
```